simapp/app.go

```diff
@@ -213,8 +211,13 @@ func NewSimApp(
 
 	var appBuilder *runtime.AppBuilder
 	var msgServiceRouter *baseapp.MsgServiceRouter
+	var txBaseAppOption func(*baseapp.BaseApp)
 
-	if err := depinject.Inject(AppConfig,
+	if err := depinject.Inject(
+		depinject.Configs(
+			AppConfig,
+			depinject.Supply(&encodingConfig.TxConfig),
```

Comment:
We shouldn't supply the TxConfig, the tx module should do that. And what's in encodingConfig should eventually be deprecated

Reply:
The issue is that there are many tests passing a custom `encodingConfig`, and those are the ones failing. Should we try to deprecate it? So we remove/refactor tests passing custom configs. (it's going to be hard, it's being used in 168 places 😓)

Reply:
Gotcha